File: inventory-project-backend/src/main/java/com/kitcheninventory/inventory_project_backend/dto/PurchaseDTO.java
package com.kitcheninventory.inventory_project_backend.dto;

import java.time.LocalDate;
import java.util.List;

public record PurchaseDTO(
    Long purchaseID,
    Double totalCost,
    LocalDate date,
    String store,
    List<PurchaseItemDTO> items
) {}

File: frontend/inventory-project/types/PurchaseDTO.ts
import { CalendarDate } from 'react-native-paper-dates/lib/typescript/Date/Calendar';
import { PurchaseItemDTO } from './PurchaseItemDTO';

export interface PurchaseDTO<T> {
    purchaseID: number | null;
    date: string;        //"2025-04-27"
    store: string;
    totalCost: number;
    items: T[];
  }